src/QMCWaveFunctions/Fermion/MultiDiracDeterminant.h

```diff
@@ -325,6 +346,20 @@ class MultiDiracDeterminant : public WaveFunctionComponent
                                           std::vector<RealType>& sign,
                                           int dx);
 
+  void mw_BuildDotProductsAndCalculateRatios(int nw,
```

Comment:
This function has the same name as the one below for value only. I think it is better to rename the function with `_Grad` to improve readability. Please update both single and multi-walker versions.